pkg/sql/logictest/tests/local/local_test: TestLogic_udf failed

pkg/sql/logictest/tests/local/local_test.TestLogic_udf [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12679058?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12679058?buildTab=artifacts#/) on release-23.1 @ [24b85a57acfca71c56c5a62ffc678a306c6e5521](https://github.com/cockroachdb/cockroach/commits/24b85a57acfca71c56c5a62ffc678a306c6e5521):


```
=== RUN   TestLogic_udf
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/ab11c9cd60408977be9520945b6e4e92/logTestLogic_udf2300512537
    test_log_scope.go:79: use -show-logs to present logs inline
    logictestbase.go:485: statement/query skipped with reason: unsupported configuration local, statement/query only supports local-legacy-schema-changer (no issue given)
=== CONT  TestLogic_udf
    logic.go:3930: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/11618/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/udf:3601: error while processing
    logic.go:3930: test was successful but validation upon completion failed: dropping database test failed: pq: polling for queued jobs to complete: poll-show-jobs: root: memory budget exceeded: 10240 bytes requested, 268431360 currently allocated, 268435456 bytes in budget
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/ab11c9cd60408977be9520945b6e4e92/logTestLogic_udf2300512537
--- FAIL: TestLogic_udf (22.73s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_udf.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/logictest/logic.go
+++ b/pkg/sql/logictest/logic.go
@@ -1424,9 +1424,16 @@ func (t *logicTest) newCluster(
 		defaultTestTenant = base.TestTenantProbabilisticOnly
 	}
 
+	// Ensure that the server has at least 320MiB of SQL memory budget. We've
+	// seen cases when the default of 256MiB is insufficient.
+	const maxSQLMemoryLimitLowerBound = 320 << 20 /* 320MiB */
+	maxSQLMemoryLimit := serverArgs.MaxSQLMemoryLimit
+	if maxSQLMemoryLimit <= maxSQLMemoryLimitLowerBound {
+		maxSQLMemoryLimit = maxSQLMemoryLimitLowerBound
+	}
 	params := base.TestClusterArgs{
 		ServerArgs: base.TestServerArgs{
-			SQLMemoryPoolSize: serverArgs.MaxSQLMemoryLimit,
+			SQLMemoryPoolSize: maxSQLMemoryLimit,
 			DefaultTestTenant: defaultTestTenant,
 			Knobs: base.TestingKnobs{
 				Store: &kvserver.StoreTestingKnobs{
@@ -4011,8 +4018,8 @@ var logicTestsConfigFilter = envutil.EnvOrDefaultString("COCKROACH_LOGIC_TESTS_C
 // want to specify for the test clusters to be created with.
 type TestServerArgs struct {
 	// MaxSQLMemoryLimit determines the value of --max-sql-memory startup
-	// argument for the server. If unset, then the default limit of 256MiB will
-	// be used.
+	// argument for the server. This value will always be bumped to at least
+	// 320MiB (including when unset).
 	MaxSQLMemoryLimit int64
 	// If set, mutations.MaxBatchSize, row.getKVBatchSize, and other values
 	// randomized via the metamorphic testing will be overridden to use the

--- a/pkg/sql/logictest/testdata/logic_test/udf
+++ b/pkg/sql/logictest/testdata/logic_test/udf
@@ -1,9 +1,3 @@
-# Set the distsql_workmem to the default production value because metamorphic
-# values may be too low for the legacy schema changer.
-onlyif config local-legacy-schema-changer
-statement ok
-SET distsql_workmem = '64MiB'
-
 statement ok
 CREATE TABLE ab (
   a INT PRIMARY KEY,

